BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    public void CopyTo(double[] array, int arrayIndex)
+    {
+        ToDoubleArray().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(decimal[] array, int arrayIndex)
+    {
+        ToDecimalArray().CopyTo(array, arrayIndex);
+    }
+
+    /// <summary>
+    /// Copy this instance to the specified array.
+    /// </summary>
+    /// <param name="arrayIndex">Start position.</param>
+    public void CopyTo(char[] array, int arrayIndex)
+    {
+        ToCharArray().CopyTo(array, arrayIndex);
+    }
+
+    public bool[] ToBooleanArray()
+    {
+        if (IsEmpty)
+            return new bool[0];
+        else
+        {
+            var bools = new bool[bits.Length];
+            for (int i = 0; i < bits.Length; i++)
+                bools[i] = bits[i];
+            return bools;
+        }
+    }
+
+    public byte[] ToByteArray()
+    {
+        if (IsEmpty)
+            return new byte[0];
+        else
+        {
+            int inum = bits.Length / 8;
+            double dnum = (double)bits.Length / 8;
+            if (dnum.ToString().Contains(".") || dnum.ToString().Contains(","))
+                inum++;
+            var bytes = new byte[inum];
+            bits.CopyTo(bytes, 0);
+            return bytes;
+        }
+    }
+
+    public short[] ToInt16Array()
+    {
+        if (IsEmpty)
+            return new short[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 2)
+                {
+                    bytesList.Add(new byte[2]);
+                    c = 0;
+                }
+                bytesList[bytesList.Count - 1][c] = bytes[i];
+            }
+            var shortList = new List<short>();
+            foreach (var shbytes in bytesList)
+                shortList.Add(BitConverter.ToInt16(shbytes, 0));
+            return shortList.ToArray();
+        }
+    }
+
+    public int[] ToInt32Array()
+    {
+        if (IsEmpty)
+            return new int[0];
+        else
+        {
+            int inum = bits.Length / 32;
+            double dnum = (double)bits.Length / 32;
+            if (dnum.ToString().Contains(".") || dnum.ToString().Contains(","))
+                inum++;
+            var ints = new int[inum];
+            bits.CopyTo(ints, 0);
+            return ints;
+        }
+    }
+
+    public long[] ToInt64Array()
+    {
+        if (IsEmpty)
+            return new long[0];
+        else
+        {
+            var bytes = ToByteArray();
+            var bytesList = new List<byte[]>();
+            var c = -1;
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                c++;
+                if (i == 0 | c == 8)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Correct this '|' to '||'.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)